src/error.rs

```diff
@@ -72,6 +72,22 @@ impl From<TemplateError> for RenderError {
     }
 }
 
+#[derive(Debug)]
+pub struct MissingVariableError(pub String);
```

Comment:
You can use `derive(Error)` to simplify this, see https://docs.rs/thiserror/latest/thiserror/